lib/house.rb

```diff
@@ -0,0 +1,75 @@
+class House
+  attr_reader :string_price,
+              :address,
+              :rooms
+  def initialize(string_price, address)
+    @string_price = string_price
+    @address = address
+    @rooms = []
+  end
+
+  def rooms_sorted_by_category
+    @rooms.group_by do |room|
+      room.category
+    end
+  end
+
+  def rooms_sorted_by_area
+    sorted = @rooms.sort_by do |room|
+      room.length * room.width.to_i
+    end
```

Comment:
You could also call .reverse just after the `end`.  And, don't forget about methods that you have already created!

```ruby
  def rooms_sorted_by_area
    @rooms.sort_by do |room|
      room.area
    end.reverse
  end
```